qiskit_quantuminspire/qi_jobs.py

```diff
@@ -222,9 +222,11 @@ async def _fetch_job_results(self) -> None:
                 circuit_data.results = None if not result_item else result_item[0]
 
     @cache
-    def result(self) -> Result:
+    def result(self, wait_for_results: Optional[bool] = True, timeout: float = 60.0) -> Result:
         """Return the results of the job."""
-        if not self.done():
+        if wait_for_results:
```

Comment:
Shouldn't we do a wait or something there? If I see this correctly, we just keep on looping, and I don't see the timeout being used somewhere?

Reply:
Is it an idea to add a backlog limit? In the e2e tests we have a max number of tries. We could add that to this function as well; for 0 (default), we keep on polling till the cows come home. for any positive integer, we poll as many times and fail afterwards.

Is that an idea, of not?